### ISSUE TITLE
Regex unicode property autocomplete

### ISSUE THREAD

**Issue by bgenia** [microsoft/TypeScript#59697]
### 🔍 Search Terms

regular expressions, unicode character class, unicode property, hints, autocomplete

### ✅ Viability Checklist

- [X] This wouldn't be a breaking change in existing TypeScript/JavaScript code
- [X] This wouldn't change the runtime behavior of existing JavaScript code
- [X] This could be implemented without emitting different JS based on the types of the expressions
- [X] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
- [X] This isn't a request to add a new utility type: https://github.com/microsoft/TypeScript/wiki/No-New-Utility-Types
- [X] This feature would agree with the rest of our Design Goals: https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals

### ⭐ Suggestion

Since TS now validates regular expressions including unicode properties and even provides spelling suggestions, why not autocomplete them?

### 📃 Motivating Example

In this example TS 5.5 understands that the first regex is incorrect:
```ts
const re1 = /\p{Emoj}/u // Error: Unknown Unicode property name or value. Did you mean "Emoji"?
const re2 = /\p{Emoji}/u // Ok
```

### 💻 Use Cases

It would be convenient if TS suggested available unicode properties when suggestions are triggered inside `\p{}`/`\P{}` sequences.
